### PR TITLE
Feature/ratings

### PR DESCRIPTION
**Pull request by tschokri**

### CODE REVIEW

File: src/ratings/ratings.module.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@Module({
+  imports: [
+    TypeOrmModule.forFeature([Rating, Wine, Winemaker, Store]),
+    WinesModule,
+    WinemakersModule,
+    StoresModule,
+  ],
+  controllers: [RatingsController],
+  providers: [RatingsService, WinesService, WinemakersService, StoresService],
```
Comment (devskar) [inline src/ratings/ratings.module.ts:24]: I just a little bit confused about all the imports/providers here. Is it required to provide all the services the WinesService uses?

**Review by devskar — CHANGES_REQUESTED** [review]
I am not sure about this approach here. The service is perfect, but i am not sure about the controller. now we have an endpoint that looks like `/api/ratings`, but I would suggest to have something like `/api/wines/${id}/ratings`and `/api/users/${name}/ratings`

**Comment by tschokri** [comment]
> I am not sure about this approach here. The service is perfect, but i am not sure about the controller. now we have an endpoint that looks like `/api/ratings`, but I would suggest to have something like `/api/wines/${id}/ratings`and `/api/users/${name}/ratings`

sounds reasonable, i'll look into it, thank you

File: src/ratings/entities/rating.entity.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  Entity,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Wine } from '../../wines/entities/wine.entity';
+import { IsInt, Max, Min } from 'class-validator';
+
+@Entity()
+export class Rating {
```
Comment (devskar) [inline src/ratings/entities/rating.entity.ts:13]: I am pretty sure we should save the author of the review. for authentication and for displaying purposes. And to also get all the ratings a user submitted

**Review by devskar — CHANGES_REQUESTED** [review]

File: src/ratings/dtos/create-rating.dto.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import { PickType } from '@nestjs/swagger';
 import { Rating } from '../entities/rating.entity';
+import { IsNotEmpty } from 'class-validator';
 
 export class CreateRatingDto extends PickType(Rating, [
   'stars',
   'text',
 ] as const) {
+  @IsNotEmpty()
```
Comment (devskar) [inline src/ratings/dtos/create-rating.dto.ts:9]: I just found out that there is something called @IsUUID. Maybe try it out and maybe directly add it to every uuid or add a comment to #39

**Review by devskar — CHANGES_REQUESTED** [review]

File: src/ratings/ratings.module.ts (hunk 1 above)
Comment (tschokri) [reply src/ratings/ratings.module.ts:24]: as far as i understand its necessary to include all services

File: src/ratings/entities/rating.entity.ts (hunk 2 above)
Comment (tschokri) [reply src/ratings/entities/rating.entity.ts:14]: done

**Review by devskar — APPROVED** [review]
Clean

**Pull request merged**